### PR TITLE
Reports were not fetched properly

### PR DESCRIPTION
**Pull request by Bluesmile82**
This should fix the fetching of the new reports into the reports page:
The reports were not checking the index provided by the url. Also changed the update of the page to fetch also when the url changes.

Due to errors the following updates have been made to bring the filter bar in sync with templates:
- Abstract all props logic to reports container and extract to funcs
- Convert current indexing or ids to real ids in params for simpler props checking
- FIX: areas (and all filters) now apply correctly to the view

ADDITIONAL: 
- loaders have been added to the table view when fetching either templates or reports.
- loader added when finding user location
- translations added

### CODE REVIEW

**Comment by edbrett** [comment]
Updates have been made. Please review. 👓

File: src/components/pages/reports/ReportsContainer.js [hunk 1]
```diff
@@ -1,83 +1,100 @@
...
-          answers = filterBy('aoi', answers, aoi);
-        }
-        if (searchValues !== undefined){ 
-          answers = filterBy('search', answers, searchValues);
-        }
-        if (date !== undefined){ 
-          answers = filterBy('date', answers, date);
-        }
-      }
+const getTemplateOptions = (templates) => {
```
Comment (j8seangel) [inline src/components/pages/reports/ReportsContainer.js:37]: Why don't do that on the `ReportsFiltersContainer` directly?
I mean we are doing in this container and then pass as props to the ReportsFilter component.
Also let you know that the filter is not being populate correctly 
image

**Review by j8seangel — CHANGES_REQUESTED** [review]
Please review the report filter by template bug.
Also, I think we should change the text of the notifications when there isn't any response because it shows `Unable to load reports` and we did correctly but they are empty.

**Comment by edbrett** [comment]
i18n string for no results in selectors added.

**Pull request merged**